atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ManyClientPoolingContainer.java

```diff
@@ -117,8 +121,8 @@ public synchronized void shutdownPooling() {
 
     @Override
     protected PoolingContainer<Client> delegate() {
-        List<PoolingContainer<Client>> list = containers;
-        return list.get(random.nextInt(list.size()));
+        ImmutableList<PoolingContainer<Client>> list = containers; // volatile read
```

Comment:
Why are we doing this? It's not a pattern I've seen elsewhere, but the comment gives me hop there's a reason.